core/system/src/mpu/vmpu.c

```diff
@@ -59,16 +59,24 @@ static int vmpu_sanity_checks(void)
     assert(vmpu_bits(0x8001UL) == 16);
     assert(vmpu_bits(1) == 1);
 
+    /* Verify that the known hard-coded symbols are equal to the ones taken from
+     * the host linker script. */
+    assert((uint32_t) __uvisor_config.flash_start == FLASH_ORIGIN);
+    assert((uint32_t) __uvisor_config.flash_offset == FLASH_ORIGIN + RESERVED_FLASH);
+    assert((uint32_t) __uvisor_config.sram_start == SRAM_ORIGIN);
+    assert((uint32_t) __uvisor_config.sram_offset == SRAM_ORIGIN + RESERVED_SRAM);
+
+    /* Verify that the uVisor binary blob is positioned at the flash offset. */
+    assert(__uvisor_config.flash_offset == __uvisor_config.main_start);
+
     /* verify that __uvisor_config is within valid flash */
-    assert( ((uint32_t) &__uvisor_config) >= FLASH_ORIGIN );
-    assert( ((((uint32_t) &__uvisor_config) + sizeof(__uvisor_config))
-             <= (FLASH_ORIGIN + FLASH_LENGTH)) );
+    assert(VMPU_FLASH_ADDR((uint32_t) &__uvisor_config));
+    assert(VMPU_FLASH_ADDR(((uint32_t) &__uvisor_config) + sizeof(__uvisor_config)));
```

Comment:
I would prefer leaving the comparison operators here rather than moving them inside the `VMPU_FLASH_ADDR` macro, or renaming the macro to `VMPU_FLASH_ADDR_VALID`.